lib/storage/src/CookiesProvider.ts

```diff
@@ -0,0 +1,38 @@
+import { Browser, Provider } from '@exteranto/support'
+import { Cookies } from './Cookies'
+import { Cookies as ChromeCookies } from './chrome/Cookies'
+import { Cookies as ExtensionsCookies } from './extensions/Cookies'
+import { Dispatcher } from '@exteranto/events'
+import { NotImplementedException } from '@exteranto/exceptions'
+
+export class CookiesProvider extends Provider {
+
+  /**
+   * Boot the provider services.
+   */
+  public boot () : void {
+    this.container.bind(ChromeCookies).to(Cookies).for(Browser.CHROME)
+    this.container.bind(ExtensionsCookies).to(Cookies).for(Browser.EXTENSIONS)
+
+    /*
+    if (Browser.SAFARI) {
```

Comment:
@kouks any ideas on a neat way to inform the developer about the issue with Safari not being implemented besides having it in the docs?

Reply:
I think we might push this functionality up to the container itself. Rather than returning `null`, we could throw or `console.warn` an exception that would say that the dependency was not found.

Reply:
^ I would be up for that. But I don't think that is enough. There should be some warning upon initiation of the service, not only when the service is actually used imo.

Reply:
Use the `if` statement for now. It is clearer than the `||` notation. I shall think about a better solution later. I want to reconsider the `only` method on providers and maybe solve this issue as by-product.

Reply:
Ok, will do. Also, I have had issues with `@HasAccessTo` annotation on `boot` method of a provider. Even though `PermissionsManagerProvider` module was in the array of providers before `CookieProvider`, the `PermissionManager` was `null` upon resolving from `Container`. Not a problem atm, just noting it.

Reply:
I thought something like this might happen, maybe create an issue and I will have a look.

Reply:
Wait `if (Browser.SAFARI) {` what?

Reply:
`if (this.container.resolveParam('browser') === Browser.SAFARI)`